azurerm/resource_arm_postgresql_database.go

```diff
@@ -119,8 +120,7 @@ func resourceArmPostgreSQLDatabaseRead(d *schema.ResourceData, meta interface{})
 	d.Set("resource_group_name", resGroup)
 	d.Set("server_name", serverName)
 	d.Set("charset", resp.Charset)
-	d.Set("collation", resp.Collation)
-
+	d.Set("collation", utils.AzureRMNormalizeCollation(d.Get("collation").(string), "-", "_", -1))
```

Comment:
providing we split this out to a separate line (to make it more readable) - we should be able to just use a string replacement directly? e.g.

```
collation := strings.Replace(d.Get(resp.Collation, "-", "_", -1)
d.Set("collation", collation)
```

note: in general we shouldn't be retrieving any properties from the schema in the Read function - since they won't exist during Import and can lead to crashes.